plugins/scaffolder-backend/src/scaffolder/stages/publish/azure.ts

```diff
@@ -15,27 +15,41 @@
  */
 
 import { PublisherBase, PublisherOptions, PublisherResult } from './types';
-import { GitApi } from 'azure-devops-node-api/GitApi';
+import { IGitApi } from 'azure-devops-node-api/GitApi';
 import { GitRepositoryCreateOptions } from 'azure-devops-node-api/interfaces/GitInterfaces';
-import { JsonValue } from '@backstage/config';
-import { RequiredTemplateValues } from '../templater';
 import { initRepoAndPush } from './helpers';
+import { AzureIntegrationConfig } from '@backstage/integration';
+import gitUrlParse from 'git-url-parse';
+import { getPersonalAccessTokenHandler, WebApi } from 'azure-devops-node-api';
 
 export class AzurePublisher implements PublisherBase {
-  private readonly client: GitApi;
-  private readonly token: string;
-
-  constructor(client: GitApi, token: string) {
-    this.client = client;
-    this.token = token;
+  static async fromConfig(config: AzureIntegrationConfig) {
+    if (!config.token) {
```

Comment:
hm maybe filter on this on the caller side instead and log a warning that publishing won't work to that host because there was no token set - and then throw in here if there's no token